### ISSUE TITLE
Crash on non-numeric coordinate

### ISSUE THREAD

**Issue by benburrill** [bulletmark/libinput-gestures#180]
```
$ libinput-gestures -l
libinput-gestures: session KDE+x11 on Linux-4.18.11-200.fc28.x86_64-x86_64-with-fedora-28-Twenty_Eight, python 3.6.6, libinput 1.11.3
Gestures configured in ~/.config/libinput-gestures.conf:
swipe up           false
swipe down         false
swipe right        _internal ws_up
swipe left         _internal ws_down
pinch in           false
pinch out          false
libinput-gestures: device /dev/input/event4: SynPS/2 Synaptics TouchPad
```

**Describe the issue**
I just ran into an issue where (by the looks of it) one of the coords in a pinch event got parsed as `'-'`, resulting in a crash (traceback below).  This has never happened before in the few months or so I've used libinput-gestures and it coincided with an intermittent and mysterious issue I've been experiencing with my touchpad recently where it starts behaving extremely erratically for some time until the haunting of my touchpad stops as mysteriously as it started.

I'm reporting this bug partly because I don't think libinput-gestures should crash in this case (I think invalid coordinates should be considered to be `0` or something), but also in the hopes you might have some insight into what this anomalous data could mean and how it might relate to my touchpad problems.

Traceback and info from ABRT:

```
Traceback (most recent call last):
  File "/usr/bin/libinput-gestures", line 543, in <module>
    handler.update(re.split(r'[^-.\d]+', params))
  File "/usr/bin/libinput-gestures", line 341, in update
    self.data[0] += float(coords[4]) - 1.0
ValueError: could not convert string to float: '-'

Local variables in innermost frame:
coords: ['0.00', '0.00', '0.00', '0.00', '-', '0.00']
self: <__main__.PINCH object at 0x7f3255f23a20>
```

**Comment by bulletmark** [comment]
Hey, a valid bug for once! Thanks ;)

Can you please run `libinput-gestures -r` on the command line and do a heap of pinch gestures hopefully to capture the same sequence which caused this?

**Comment by benburrill** [comment]
I wasn't able to reproduce it again during the period my touchpad remained haunted (actually looking at ABRT it looks like it did crash a second time, but KDE didn't notify me for some reason, so I wasn't able to interrupt `libinput-gestures -r` in time), but I'll try again next time (probably combined with a better system for catching the problem when it occurs).

**Comment by bulletmark** [comment]
You can't run a second copy in raw mode anyhow. I have added commit 9cdc78395659d9d8aec9d350463b3d15dd7c1db6 to protect against this issue and any other parsing errors in general which I should have been doing anyhow. Now the program will simply ignore that libinput update message and will output a log error line showing the data which could not be parsed. So please run this version and see if you can give me some of the sample error lines.

**Comment by benburrill** [comment]
I see, thanks for the info and fix.  I have upgraded and have things set up so it should log "Could not parse" errors for me.  I'll let you know what I find.

**Comment by bulletmark** [comment]
@benburrill have you looked through your journal to see if there are any parse errors?

**Comment by benburrill** [comment]
Nothing yet, but I think the weird data is associated with the intermittent touchpad problems I have.  Next time I experience these problems, I'll play around with the touchpad a bunch and hopefully reproduce the issue.

**Comment by bulletmark** [comment]
I've released version 2.39 with the change to prevent this problem so will close this bug. Please come back and paste the error message if you see it eventually.
